code/src/main/java/com/google/sps/storage/data-storage/PositiveUser.java

```diff
@@ -0,0 +1,60 @@
+package com.google.sps.storage;
+
+import com.googlecode.objectify.annotation.Entity;
+import com.googlecode.objectify.annotation.Id;
+import java.time.Instant;
+
+/**
+* Store the information associated with positive user.
+*/
+@Entity
+public class PositiveUser {
+
+  @Id private String userId;
+  private String userEmail;
+  private int emailsSent;
+  private long firstLoginSeconds;
+  private long lastLoginSeconds;
+
+  // Objecify requires one constructor with no parameters
+  private PositiveUser() {}
+  
+  public PositiveUser(String id, String email) {
+    userId = id;
+    userEmail = email;
+    emailsSent = 0;
+    firstLoginSeconds = lastLoginSeconds = Instant.now().getEpochSecond();
+  }
+
+  public void setLastLogin() {
+    firstLoginSeconds = Instant.now().getEpochSecond();
+  }
+  
+  public String getUserId() {
+      return userId;
+  }
+
+  public String getUserEmail() {
+    return userEmail;
+  }
+  
+  public long getNumberOfEmailsSent() {
+      return emailsSent;
+  }
+
+  public long getFirstLoginInSeconds() {
+    return firstLoginSeconds;
+  }
+
+  public long getLastLoginInSeconds() {
+    return lastLoginSeconds;
+  }
+
+  public void incrementEmailsSent() {
+    emailsSent++;
+  }
+
+  public boolean emailLimitReached() {
```

Comment:
As it is this seems to actually be the opposite (userCanStillSendEmails) rather than (userCanNoLongerSendEmails = limitReached). Please name a little bit more precicely.

Reply:
done